packages/smart-contracts/test/contracts/BatchEthPayments.test.ts

```diff
@@ -0,0 +1,321 @@
+          [10, 20],
+          feeAddress,
+          {
+            value: BigNumber.from('1000'),
+          },
+        ),
+    )
+      .to.emit(ethFeeProxy, 'TransferWithReferenceAndFee')
+      .withArgs(payee1, '200', ethers.utils.keccak256(referenceExample1), '10', feeAddress)
+      .to.emit(ethFeeProxy, 'TransferWithReferenceAndFee')
+      .withArgs(payee2, '300', ethers.utils.keccak256(referenceExample2), '20', feeAddress);
+
+    afterEthBalance1 = await provider.getBalance(payee1);
+    expect(afterEthBalance1).to.be.equal(beforeEthBalance1.add(200));
+
+    afterEthBalance2 = await provider.getBalance(payee2);
+    expect(afterEthBalance2).to.be.equal(beforeEthBalance2.add(300));
+
+    expect(await provider.getBalance(batchAddress)).to.be.equal(0);
+  });
+
+  it('Should pay 2 payments with the exact amount', async function () {
+    beforeEthBalance1 = await provider.getBalance(payee1);
+    beforeEthBalance2 = await provider.getBalance(payee2);
+
+    const totalAmout = BigNumber.from('535'); // amount: 500, fee: 10+20, batchFee: 2+3
+
+    const tx = await batch
+      .connect(owner)
+      .batchEthPaymentsWithReference(
+        [payee1, payee2],
+        [200, 300],
+        [referenceExample1, referenceExample2],
+        [10, 20],
+        feeAddress,
+        {
+          value: totalAmout,
+        },
+      );
+    await tx.wait();
+
+    afterEthBalance1 = await provider.getBalance(payee1);
+    expect(afterEthBalance1).to.be.equal(beforeEthBalance1.add(200));
+
+    afterEthBalance2 = await provider.getBalance(payee2);
+    expect(afterEthBalance2).to.be.equal(beforeEthBalance2.add(300));
+
+    expect(await provider.getBalance(batchAddress)).to.be.equal(0);
+  });
+
+  it('Should pay 10 Ethereum payments', async function () {
+    beforeEthBalance2 = await provider.getBalance(payee2);
+
+    const amount = 2;
+    const feeAmount = 1;
+    const nbTxs = 10; // to compare gas optim, go to 100.
+    const [_, recipients, amounts, paymentReferences, feeAmounts] = getBatchPaymentsInputs(
+      nbTxs,
+      '_noTokenAddress',
+      payee2,
+      amount,
+      referenceExample1,
+      feeAmount,
+    );
+    const totalAmount = BigNumber.from(((amount + feeAmount) * nbTxs).toString());
+
+    const tx = await batch
+      .connect(owner)
+      .batchEthPaymentsWithReference(
+        recipients,
+        amounts,
+        paymentReferences,
+        feeAmounts,
+        feeAddress,
+        {
+          value: totalAmount,
+        },
+      );
+
+    const receipt = await tx.wait();
+    if (logGasInfos) {
+      console.log(`nbTxs= ${nbTxs}, gas consumption: `, receipt.gasUsed.toString());
+    }
+
+    afterEthBalance2 = await provider.getBalance(payee2);
+    expect(afterEthBalance2).to.be.equal(beforeEthBalance2.add(amount * nbTxs));
+
+    expect(await provider.getBalance(batchAddress)).to.be.equal(0);
+  });
+
+  it('Should revert batch if not enough funds', async function () {
```

Comment:
It's wise to nest the former test cases under a describe "happy path" for example, and starting from this test case, nest under a describe "expected reverts".